internal/pkg/cli/deploy/workload_test.go

```diff
@@ -211,7 +211,7 @@ func TestWorkloadDeployer_UploadArtifacts(t *testing.T) {
 						"com.aws.copilot.image.builder":        "copilot-cli",
 						"com.aws.copilot.image.container.name": "mockWkld",
 					},
-				}).Return("", mockError)
+				}, gomock.Any()).Return("", mockError)
```

Comment:
I wonder if the difficulty in writing tests for this logic is a hint that we should be extracting our `buildPushOutputBuffer`, `terminalBuildPushPrinter`, `ciBuildPushPrinter` into its own package that would allow us to unit test it as well as mock the printers 💭 

Perhaps, having a separate package under `term/` would help something like:
```go
package term/streambuffer  // A new streambuffer package

type StreamBuffer struct // replaces buildPushOutputBuffer as a more generic option.
type TermBufferPrinter struct // replaces terminalBuildPushPrinter.
type CIBufferPrinter struct // replaces ciBuildPushPrinter.

func PrintAll(printers []StreamBufferPrinter) error

func (tb *TermBuffer) TermPrinter(fw FileWriter) *TermBufferPrinter
func (tb *TermBuffer) CIPrinter(w io.writer) *CIPrinter
```

Reply:
Created a `syncbuffer` package for all the Buffers and `TermPrinter`.